README.md

```diff
@@ -5,757 +5,875 @@
-//            name: 'Mr Griffin Senior'
-//          }
-//        }
-//      }
-//    }
+let blog = create(Blog, {
+  posts: [
+    { id: 1, title: "Hello World" },
+    { id: 2, title: "Most fascinating blog in the world" }
+  ]
+});
+
+blog.posts[0].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts[1].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
 ```
 
-We currently do not support composition inside of arrays but we're looking into supporting it in the
-future.
+When you're working with an array microstate, the shape of the Microstate is determined by the value. In this case, `posts` is created with two items which will, in turn, create a Microstate with two items. Each item will be a Microstate of type `Post`. If you push another item onto the `posts` Microstate, it'll be treated as a `Post`.
 
-Composed states have two default transitions `set` and `merge`.
+```js
+let blog2 = blog.posts.push({ id: 3, title: "It is only getter better" });
+
+blog2.posts[2].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
+```
 
-`set` will replace the state of the current microstate.
+Notice how we didn't have to do any extra work to define the state transition of adding another post to the list? That's the power of composition!
+
+## Object Microstates
+
+You can also create an object microstate with `{Post}`. The difference is that the collection is treated as an object. This can be helpful when create normalized data stores.
 
 ```js
-create(Person).father.father.set({ name: 'Peter' }).state;
-// => { name: '', father: { name: 'Peter' }}
+class Blog {
+  posts = { Post };
+}
+
+class Post {
+  id = Number;
+  title = String;
+}
+
+let blog = create(Blog, {
+  posts: {
+    "1": { id: "1", title: "Hello World" },
+    "2": { id: 2, title: "Most fascinating blog in the world" }
+  }
+});
+
+blog.posts["0"].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts["1"].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
 ```
 
-`merge` will recursively merge the object into the current state.
+Object type microstates have `Object` transitions, such as `assign`, `put` and `delete`.
 
 ```js
-create(Person, { name: 'Peter' }).merge({ name: 'Taras', father: { name: 'Serge' } }).state;
-// { name: 'Taras', father: { name: 'Serge' }}
+let blog2 = blog.posts.put("3", { id: 3, title: "It is only getter better" });
+
+blog2.posts["3"].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
 ```
 
-## Static values
+# Transitions
+
+Transitions are the operations that let you derive a new state from an existing state. All transitions return another Microstate. You can use state charts to visualize microstates. The `Boolean` type can be described with the following statechart. 
 
-You can define static values on custom types. Static values are added to state but do not get
-transitions.
+<img src="README/boolean-statechart.png" alt="Boolean Statechart" width="250" />
```

Comment:
This is still to small to read:

![image](https://user-images.githubusercontent.com/4205/41997304-d2874b6e-7a14-11e8-9d3b-898dbda1e82d.png)

Can we put a caption on this?

Reply:
I don't think so. I made the image 2x bigger.